plugin/trino-ignite/src/main/java/io/trino/plugin/ignite/IgniteClient.java

```diff
@@ -0,0 +1,595 @@
+        }
+        catch (SQLException e) {
+            throw new TrinoException(JDBC_ERROR, e);
+        }
+    }
+
+    public Map<String, Object> getTableProperties(Connection connection, JdbcTableHandle tableHandle)
+    {
+        ImmutableMap.Builder<String, Object> properties = ImmutableMap.builder();
+        SchemaTableName schemaTableName = tableHandle.asPlainTable().getSchemaTableName();
+        String tableName = requireNonNull(schemaTableName.getTableName(), "Ignite table name can not be null").toUpperCase(ENGLISH);
+        String sql = format("SELECT idx.CACHE_ID, " +
+                "(SELECT COLUMNS FROM sys.indexes WHERE SCHEMA_NAME = 'PUBLIC' AND TABLE_NAME = '%s' AND INDEX_NAME = '_key_PK') AS PKS," +
+                "(SELECT COLUMNS FROM sys.indexes WHERE SCHEMA_NAME = 'PUBLIC' AND TABLE_NAME = '%1$s' AND INDEX_NAME = 'AFFINITY_KEY') AS AFK FROM sys.indexes as idx " +
+                "JOIN sys.caches che ON idx.CACHE_ID = che.CACHE_ID WHERE idx.SCHEMA_NAME = 'PUBLIC' AND idx.TABLE_NAME = '%1$s' LIMIT 1", tableName);
+
+        try (PreparedStatement preparedStatement = connection.prepareStatement(sql);
+                ResultSet resultSet = preparedStatement.executeQuery()) {
+            if (!resultSet.next()) {
+                return ImmutableMap.of();
+            }
+            List<String> primaryKeys = extractColumnNamesFromOrderingScheme(resultSet.getString("PKS"));
+            checkArgument(!primaryKeys.isEmpty(), "Ignite table should has at least one primary key");
+            properties.put(PRIMARY_KEY_PROPERTY, primaryKeys);
+        }
+        catch (SQLException e) {
+            throw new TrinoException(JDBC_ERROR, e);
+        }
+        return properties.buildOrThrow();
+    }
+
+    // extract result from format: "ID" ASC, "CITY_ID" ASC
+    private List<String> extractColumnNamesFromOrderingScheme(String row)
+    {
+        ImmutableList.Builder<String> builder = ImmutableList.builder();
+        if (isNullOrEmpty(row)) {
+            return builder.build();
+        }
+
+        List<String> fields = SPLITTER.splitToList(row);
+        for (int i = 0; i < fields.size(); i += 2) {
+            String field = fields.get(i);
+            checkArgument(!isNullOrEmpty(field), "Ignite column name is empty");
+            builder.add(field.toLowerCase(ENGLISH));
+        }
+
+        return builder.build();
+    }
+
+    @Override
+    public String buildInsertSql(JdbcOutputTableHandle handle, List<WriteFunction> columnWriters)
+    {
+        IgniteOutputTableHandle outputHandle = (IgniteOutputTableHandle) handle;
+        String params = columnWriters.stream()
+                .map(WriteFunction::getBindExpression)
+                .collect(joining(","));
+        String columns = handle.getColumnNames().stream().map(this::quoted).collect(joining(", "));
+        if (outputHandle.dummyIdColumn().isPresent()) {
+            String nextId = "CAST(UUID() AS VARCHAR), ";
```

Comment:
I'm little wondering why we are casting to varchar. It looks redundant to me. I would recommend changing to uuid type or leaving a comment if there's a reason doing so.

Reply:
Currently, we do not support `uuid` type in Ignite yet.